crypto/evp/evp_rand.c

```diff
@@ -680,3 +692,60 @@ int EVP_RAND_verify_zeroization(EVP_RAND_CTX *ctx)
     evp_rand_unlock(ctx);
     return res;
 }
+
+int evp_rand_can_seed(EVP_RAND_CTX *ctx)
+{
+    return ctx->meth->get_seed != NULL;
+}
+
+static size_t evp_rand_get_seed_locked(EVP_RAND_CTX *ctx,
+                                       unsigned char **buffer,
+                                       int entropy,
+                                       size_t min_len, size_t max_len,
+                                       int prediction_resistance,
+                                       const unsigned char *adin,
+                                       size_t adin_len)
+{
+    if (ctx->meth->get_seed != NULL)
+        return ctx->meth->get_seed(ctx->algctx, buffer,
+                                   entropy, min_len, max_len,
+                                   prediction_resistance,
+                                   adin, adin_len);
+    return 0;
+}
+
+
+size_t evp_rand_get_seed(EVP_RAND_CTX *ctx,
```

Comment:
Not a public API (yet). Could be made public as `EVP_RAND_get_seed()` in 3.2 and backported as non-public function.

Reply:
Wasn't intended to be EVP visible but the underlying calls are part of our public provider API.

I think there are a number of risks associated with making this public.  Chief among them, misuse of the material.

Reply:
So why is `EVP_rand()` public and `EVP_RAND_get_seed()` not?

Reply:
`EVP_RAND()`?? It's a type not a function.

The seeding is internal to the DRBGs -- never called by anything else.  The DRBG's know about a few of their parent's calls and go direct so that they can support cross provider chaining.

At least before this PR it was internal.

Reply:
Sorry, I meant to write `EVP_RAND_nonce()`.